[Getting Started] Add file paths to code blocks [ci skip]

harsh183 opened the pull request:
<!--
Thanks for contributing to Rails!

Please do not make *Draft* pull requests, as they still send
notifications to everyone watching the Rails repo.

Create a pull request when it is ready for review and feedback
from the Rails team :).

If your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

About this template

The following template aims to help contributors write a good description for their pull requests.
We'd like you to provide a description of the changes in your pull request (i.e. bugs fixed or features added), the motivation behind the changes, and complete the checklist below before opening a pull request.

Feel free to discard it if you need to (e.g. when you just fix a typo). -->

### Motivation / Background

<!--
Describe why this Pull Request needs to be merged. What bug have you fixed? What feature have you added? Why is it important?
If you are fixing a specific issue, include "Fixes #ISSUE" (replace with the issue number, remove the quotes) and the issue will be linked to this PR.
-->
I talked to some people who did the tutorial and found that they were finding the blocks confusing since the filename was not always mentioned, especially later in the tutorial. For someone familiar with rails and path conventions its easy to find the way back so I didn't notice this at first but I see how this can helped. After talking to @rafaelfranca  on the rails discord I decided to explore it.

### Detail

* Added filepath with comments at the top of each block as per
  convention I've seen online
* Incremented line numbers accordingly so that the offset is right.

In some cases it does feel a bit redundant early on since we are quite explicit about which file, but it does get more useful as the tutorial goes on and following the convention consistently is useful.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature. - N/A
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included. - N/A

harsh183 commented on guides/source/getting_started.md:
````diff
@@ -907,6 +923,7 @@ The `index` action will render `app/views/products/index.html.erb`. If we open
 up that file in our code editor, we'll see the HTML it renders.
 
 ```erb
+<!-- app/views/products/index.html.erb -->
````
I never thought about using ERB this way, this is great. I'll change the PR accordingly

rafaelfranca commented on guides/source/getting_started.md:
````diff
@@ -1390,7 +1423,8 @@ To learn more about view partials, check out the
 
-```erb#4
+```erb#5
+<!-- app/views/products/show.html.erb -->
````
ERB comment here

rafaelfranca reviewed (approved):
I prefer the comment version since it is used everywhere in the guides and make is possible for us to use the same code block for multiple files if that usage make sense somewhere in the guide, but I'll let @p8 and @AmandaPerino decide.

harsh183 committed:
[Getting Started] Add file paths to code blocks

I talked to some people who did the tutorial and found that they were
finding the blocks confusing since the filename was not always
mentioned, especially later in the tutorial. For someone familiar with
rails and path conventions its easy to find the way back so I didn't
notice this at first but I see how this can helped. After talking to
refaelfranca on the rails discord I decided to explore it.

harsh183 commented:
the merge conflicts are fixed but I have no way to remove the labels that were automatically added. Lmk if I can do something else and apologies for the `rebase` mess.
